app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -30,6 +30,42 @@ def supports_update?
       true
     end
 
+    def supports_operating_systems?
+      client.respond_to?(:operating_systems) || client.send(:client).respond_to?(:operating_systems)
+    end
+
+    def determine_os_type(host)
```

Comment:
What would be a benefit of doing so? The responsibility of the oVirt model is preparing the layer between Foreman and Ovirt. Extracgint this into a separate concern just adds one more file to look when you search on what happens when you try to create some host on some compute resource. I've not seen any other concern that would do something similar (and not should I would be glad to see one)